libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -3496,15 +3496,26 @@ - (CTTelephonyNetworkInfo *)telephonyNetworkInfo {
 }
 
 - (NSString *)countryCodeByCarrier {
-  NSString *isoCode = [[self.telephonyNetworkInfo subscriberCellularProvider] isoCountryCode];
-
-  // The 2nd part of the if is working around an iOS 7 bug
-  // If the SIM card is missing, iOS 7 returns an empty string instead of nil
-  if (isoCode.length == 0) {
-    isoCode = NB_UNKNOWN_REGION;
-  }
-
-  return isoCode;
+#if __IPHONE_OS_VERSION_MIN_REQUIRED >= 120000
+    NSDictionary<NSString *, CTCarrier *> *serviceSubscriberCellularProviders = [self.telephonyNetworkInfo serviceSubscriberCellularProviders];
+    CTCarrier *carrier = serviceSubscriberCellularProviders.allValues.firstObject;
```

Comment:
Is there a reason to limit the search to only the first carrier?

I'd look through all the entries, taking the first non-empty country code.

Reply:
Thanks @paiv , I'll have time to test this out tomorrow.